app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -2547,1772 +2569,215 @@ describe('NetworkController', () => {
           const fakeNetworkClient = buildFakeClient(fakeProvider);
           mockCreateNetworkClient()
             .calledWith({
-              rpcUrl: 'https://mock-rpc-url-2',
-              chainId: '0x222',
+              rpcUrl: 'https://mock-rpc-url',
+              chainId: toHex(111),
               type: NetworkClientType.Custom,
             })
             .mockReturnValue(fakeNetworkClient);
 
           await controller.setActiveNetwork('testNetworkConfiguration');
 
           expect(controller.store.getState().providerConfig).toStrictEqual({
-            id: 'testNetworkConfiguration',
             type: 'rpc',
-            rpcUrl: 'https://mock-rpc-url-2',
-            chainId: '0x222',
-            ticker: 'TEST2',
-            nickname: 'test network 2',
+            rpcUrl: 'https://mock-rpc-url',
+            chainId: toHex(111),
+            ticker: 'TEST',
+            nickname: 'something existing',
+            id: 'testNetworkConfigurationId',
             rpcPrefs: {
               blockExplorerUrl: 'https://test-block-explorer-2.com',
             },
           });
         },
       );
     });
+  });
 
-    it('emits networkWillChange before making any changes to the network status', async () => {
-      await withController(
-        {
-          state: {
-            providerConfig: {
-              type: 'rpc',
-              rpcUrl: 'https://mock-rpc-url-1',
-              chainId: '0x111',
-              ticker: 'TEST1',
-            },
-            networkConfigurations: {
-              testNetworkConfiguration: {
-                id: 'testNetworkConfiguration',
-                rpcUrl: 'https://mock-rpc-url-2',
-                chainId: '0x222',
-                ticker: 'TEST2',
-              },
-            },
+  describe('setProviderType', () => {
+    for (const {
+      networkType,
+      chainId,
+      ticker,
+      blockExplorerUrl,
+    } of INFURA_NETWORKS) {
+      describe(`given a type of "${networkType}"`, () => {
+        refreshNetworkTests({
+          expectedProviderConfig: buildProviderConfig({
+            type: networkType,
+          }),
+          operation: async (controller) => {
+            await controller.setProviderType(networkType);
           },
-        },
-        async ({ controller, messenger }) => {
-          const fakeProviders = [
-            buildFakeProvider([
-              {
-                request: {
-                  method: 'net_version',
-                },
-                response: SUCCESSFUL_NET_VERSION_RESPONSE,
-              },
-            ]),
-            buildFakeProvider([
-              {
-                request: {
-                  method: 'net_version',
+        });
+
+        it(`overwrites the provider configuration using type: "${networkType}", chainId: "${chainId}", ticker "${ticker}", and blockExplorerUrl "${blockExplorerUrl}", clearing rpcUrl and nickname`, async () => {
```

Comment:
This test also doesn't exist on the core side. I opted to leave it because I didn't know if we wanted to bring it over to the core side or not.